### PR TITLE
The great refactor: changing the data format

### PR DESCRIPTION
**Pull request by YRMYJASKA**
Change the GPU stats we send from satellites and separate between static and dynamic data. Requires a LOT of refactoring.

### CODE REVIEW

**Review by danielg0 — APPROVED** [review]

**Comment by YRMYJASKA** [comment]
@danielg0 This should be ready for you to work on the DB refactoring.

File: cmd/satellite/main.go
Comment (aDotInTheVoid) [inline cmd/satellite/main.go]: Why are we making this change here?!?

This should be a seperate PR

File: internal/gpustats/gpustats.go
Comment (aDotInTheVoid) [inline internal/gpustats/gpustats.go:13]: NIT: What, what are the sematics of this function suposed to be? When would it be used?

I'm pritty sure adding 2 70C gpu's should make a 140C gpu.

Probably best to remove at some point, as I don't think it get's used.

File: internal/gpustats/gpustats.go
Comment (aDotInTheVoid) [inline internal/gpustats/gpustats.go:36]: NIT: Ditto

File: internal/gpustats/gpustats.go
Comment (aDotInTheVoid) [inline internal/gpustats/gpustats.go]: NIT: If we're not going to be using any of the args, we can just remove this function (and churn the tests)

File: internal/uplink/gpustats.go
Comment (aDotInTheVoid) [inline internal/uplink/gpustats.go:7]: NIT: Now that we don't share these with the frontend, we could just remove these json tags.

**Review by aDotInTheVoid — COMMENTED** [review]
Seems generally good. I don't know enough about DB stuff to review that.

As always, `NIT: ` comments don't need to be fixed before merging, but I'd like to land the singnal handling stuff seperatly, as it deserves consideration, but I don't want to block this on that

File: cmd/satellite/main.go
Comment (YRMYJASKA) [reply cmd/satellite/main.go]: Got carried away. Just a small thing that I wanted, but can remove if you want to.

File: internal/gpustats/gpustats.go
Comment (YRMYJASKA) [reply internal/gpustats/gpustats.go]: yeah, agreed. I was just mindlessly refactoring this bit. Will remove this func.

File: cmd/satellite/main.go
Comment (aDotInTheVoid) [reply cmd/satellite/main.go]: Yeah, let's please deal with this seperatly.

File: internal/uplink/gpustats.go
Comment (YRMYJASKA) [reply internal/uplink/gpustats.go:7]: Maybe, let's see how the refactors actually impact the front-end first.

**Comment by danielg0** [comment]
Now this is at a point where it compiles, I think we should consider continuing to use what's currently marked as `OldGPUStatSample` for the frontend and altering the database interface to return `workStationData`, rather than GpuStatsUpload (which, just being two slices of info and stats, has a whole bunch of assumptions the current code depends on - different number of items in each list, items at the same index being different GPUs, whether or not one is missing depending on the type of request - making the safest way to write this to be a reimp of an inner-join).

Ideally, I think we'd remove GpuStatsUpload, but we'd need to find some way of carrying around a hostname with a request without needing it in the struct the GpuStatsProvider returns. Perhaps we could add it to the headers on the request?

Then the database consumes the info on the start-up context push, and the stats whenever they come in, then on request spits out an-already zipped and grouped array (or map?) of hostnames and/to `[]OldGPUStatSample`, which is passed through to the frontend on a `all` call. Then in future if we want to add a per-machine info endpoint, we can use the same `OldGPUStatSample`, just only return one rather than an array.

**Comment by danielg0** [comment]
Also, I think it would give us the possibility to streamline some of the struct names.

Why do we have a `GPUStatSample` but the context is just `GPUInfo` (ie. why not `GPUStat`?)

**Comment by YRMYJASKA** [comment]
> Also, I think it would give us the possibility to streamline some of the struct names.
> 
> Why do we have a `GPUStatSample` but the context is just `GPUInfo` (ie. why not `GPUStat`?)

these can be changed at will later. Feel free to invent a better name.

File: internal/gpustats/gpustats.go
Comment (danielg0) [reply internal/gpustats/gpustats.go:13]: I believe @Euphoride added it back in (https://github.com/gpuctl/gpuctl/commit/325072c55246cb373d2a03f7f319d330193d05f4), so this isn't new, but my understanding is with the satellite locally storing a backlog of status packets and only sending only every so often (#68), we want a way to aggregate all the packets together (by summing so we can divide and average, etc.) into the one packet we then send off

**Pull request merged**
